docs/source/manual/snakemake.md

```diff
@@ -1,29 +1,97 @@
+
+Recall a snakemake project consists of a `Snakefile`, which describes workflow rules using an ["extension"](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html) of Python, and associated python code imported and called by these rules in the `Snakefile`.To make this project compatible with Latch, we need to do the following:
+
+1. Identify and construct explicit parameters for each file dependency in `latch_metadata.py`
+2. Build a container with all runtime dependencies
+3. Ensure your `Snakefile` is compatible with cloud execution
+
+### Constructing a `latch_metadata.py` file
+
+The snakemake framework was designed to allow developers to both define and execute their workflows. This often means that the workflow parameters are sometimes ill-defined and scattered throughout the project as configuration values, static values in the `Snakefile` or command line flags.
+
+To construct a graphical interface from a snakemake workflow, the parameters need to be explicitly identified and defined so that they can be presented to scientist to be filled out through a web application. The `latch_metadata.py` file holds these parameter definitions, along with any styling or cosmetic modifications the developer wishes to make to each parameter.
+
+Asking the question, "what are the minimum set of files needed to run my workflow to completion" is a good way to begin identifying what these parameters might be. 
+
+An example of this `latch_metadata.py` file:
+
+```
+# latch_metadata.py
+
+from pathlib import Path
+
+from latch.types.directory import LatchDir
+from latch.types.file import LatchFile
+from latch.types.metadata import LatchAuthor, SnakemakeFileParameter, SnakemakeMetadata
+
+SnakemakeMetadata(
+    display_name="fgbio Best Practise FASTQ -> Consensus Pipeline",
+    author=LatchAuthor(
+        name="Fulcrum Genomics",
+    ),
+    parameters={
+        "r1_fastq": SnakemakeFileParameter(
+            display_name="Read 1 FastQ",
+            type=LatchFile,
+            path=Path("r1.fq.gz"),
+        ),
+        "r2_fastq": SnakemakeFileParameter(
+            display_name="Read 2 FastQ",
+            type=LatchFile,
+            path=Path("r2.fq.gz"),
+        ),
+        "genome": SnakemakeFileParameter(
+            display_name="Reference Genome",
+            type=LatchDir,
+            path=Path("hs38DH"),
+        ),
+    },
+)
+```
+
+### Defining all dependencies in a container
+
+When snakemake jobs are executed on Latch, they will run in an environment defined by a `Dockerfile`. Any dependencies, such as third-party binaries, python libraries, shell scripts, need to be installed and configured correctly in this `Dockerfile` so the job has access to them.
+
+To generate a `Dockerfile` that can be modified, the following command is available:
+
+`latch dockerfile <workflow_folder>`
```

Comment:
This will not work because the autogenerated Dockerfile misses the line: `copy .latch/snakemake_jit_entrypoint.py /root/snakemake_jit_entrypoint.py` 

Bug for @maximsmol